lua/entities/sammyservers_textscreen/cl_init.lua

```diff
@@ -1,64 +1,58 @@
 include("shared.lua")
 ENT.RenderGroup = RENDERGROUP_BOTH
 
-local render_range = CreateClientConVar("ss_render_range", 1500, true, false, "Determines the render range for Textscreens. Default 1500")
-
-for i = 1, 100 do
-	surface.CreateFont("CV" .. tostring(i), {
-		font = "coolvetica",
-		size = i,
-		weight = 400,
-		antialias = false,
-		outline = true
-	})
+local render_convar = CreateClientConVar("ss_render_range", 1500, true, false, "Determines the render range for Textscreens. Default 1500")
+local render_range = render_convar:GetInt() * render_convar:GetInt()
+local textscreenFonts = textscreenFonts
+local screenInfo = {}
+
+cvars.AddChangeCallback("ss_render_range", function(convar_name, value_old, value_new)
+	render_range = tonumber(value_new) * tonumber(value_new)
+end)
+
+local function validFont(f)
+	if textscreenFonts[f] != nil then
+		return textscreenFonts[f]
+	elseif table.HasValue(textscreenFonts, f) then
+		return f
+	else
+		return false
+	end
 end
 
 function ENT:Initialize()
 	self:SetMaterial("models/effects/vol_light001")
 	self:SetRenderMode(RENDERMODE_TRANSALPHA)
 	self:SetColor(Color(255, 255, 255, 1))
 	self.lines = self.lines or {}
+//	screenInfo[self] = screenInfo[self] or {}
 	net.Start("textscreens_download")
 	net.WriteEntity(self)
 	net.SendToServer()
 end
 
 function ENT:Draw()
-	if self:GetPos():Distance(LocalPlayer():GetPos()) < render_range:GetInt() then
+	if self:GetPos():DistToSqr(LocalPlayer():GetPos()) < render_range and screenInfo[self] != nil then
```

Comment:
Nice catch. Should probably throw in a comment on why the render_range is multiplied earlier (it's easy to gloss over the change to DistToSqr, and to not know that it is more efficient than Distance)